src/OpenTelemetry/Trace/ActivitySourceAdapter.cs

```diff
@@ -84,47 +84,59 @@ internal void UpdateProcessor(ActivityProcessor processor)
 
         private void RunGetRequestedData(Activity activity)
         {
-            ActivityContext parentContext;
-            if (string.IsNullOrEmpty(activity.ParentId))
+            if (this.sampler is AlwaysOnSampler)
             {
-                parentContext = default;
+                activity.IsAllDataRequested = true;
+                activity.ActivityTraceFlags |= ActivityTraceFlags.Recorded;
             }
-            else if (activity.Parent != null)
+            else if (this.sampler is AlwaysOffSampler)
```

Comment:
Is it possible to hardwire this piece so we only check `this.sampler is AlwaysOnSampler` once and save all subsequent checks entirely?

Reply:
@cijothomas probably check with @rajkumar-rangaraj.

Reply:
@reyang Made some improvements. Let me know if this looks good. I will createbenchmark and try out later.